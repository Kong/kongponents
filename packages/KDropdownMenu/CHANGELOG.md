# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 6.36.1 (2022-08-05)


### Bug Fixes

* **kicon:** fix styling in md file ([#741](https://github.com/Kong/kongponents/issues/741)) ([e2beb71](https://github.com/Kong/kongponents/commit/e2beb71e0d406123081fe1b55b3c875f41b778d5))





# 6.36.0 (2022-08-04)


### Features

* **kdropdownmenu:** new component [khcp-690] ([#724](https://github.com/Kong/kongponents/issues/724)) ([aa37d51](https://github.com/Kong/kongponents/commit/aa37d51beaab8fef936aa9edbc6bac25b5017066))
