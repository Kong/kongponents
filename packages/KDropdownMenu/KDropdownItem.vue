<template>
  <li
    :class="{
      'has-divider': type !== 'link' && hasDivider,
      'disabled': type === 'default' && disabled,
      'danger': isDangerous,
      'k-dropdown-selected-option': selected
    }"
    :data-testid="`k-dropdown-item-${label.replace(' ', '-')}`"
    class="k-dropdown-item w-100"
  >
    <router-link
      v-if="type === 'link' && to"
      :to="!disabled ? to : $router.currentRoute.path"
      :class="{ 'disabled': disabled, 'has-divider': hasDivider }"
      class="k-dropdown-item-trigger"
      data-testid="k-dropdown-item-trigger"
    >
      <slot>{{ label }}</slot>
    </router-link>
    <KButton
      v-else-if="type === 'button'"
      :disabled="disabled"
      :is-rounded="false"
      class="k-dropdown-item-trigger btn-link k-button non-visual-button"
      data-testid="k-dropdown-item-trigger"
      v-on="listeners"
      @click="handleClick"
    >
      <slot>{{ label }}</slot>
    </KButton>
    <div
      v-else
      class="k-dropdown-item-trigger"
      data-testid="k-dropdown-item-trigger"
    >
      <slot>{{ label }}</slot>
    </div>
  </li>
</template>

<script>
import KButton from '@kongponents/kbutton/KButton.vue'

export default {
  name: 'KDropdownItem',
  components: { KButton },
  props: {
    item: {
      type: Object,
      default: null,
      // Items must have a label
      validator: (item) => item.label !== undefined
    },
    /**
     * Use this prop to add a divider above the item.
     */
    hasDivider: {
      type: Boolean,
      default: false
    },
    isDangerous: {
      type: Boolean,
      default: false
    },
    disabled: {
      type: Boolean,
      default: false
    },
    selected: {
      type: Boolean,
      default: false
    },
    /**
     * Internal use only - for tracking selection in conjunction with items prop.
     */
    selectionMenuChild: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    type () {
      if (this.item && this.item.to) {
        return 'link'
      } else if (this.$listeners.click || this.selectionMenuChild) {
        // checking $listeners since we deleted click from listeners
        return 'button'
      }

      return 'default'
    },
    label () {
      return (this.item && this.item.label) || ''
    },
    to () {
      return (this.item && this.item.to) || undefined
    },
    listeners () {
      const listeners = { ...this.$listeners }

      // use @click in template and emit
      delete listeners['click']

      return listeners
    }
  },
  methods: {
    handleClick (evt) {
      this.$emit('click', evt)

      if (this.selectionMenuChild) {
        this.$emit('change', this.item)
      }
    }
  }
}
</script>

<style lang="scss" scoped>
@import '~@kongponents/styles/variables';

li.k-dropdown-item {
  display: flex;
  align-items: center;
  font-size: 1rem;
  line-height: 1;

  &.has-divider {
    --KDropdownItemDividerContainerHeight: var(--spacing-md);
    --KDropdownItemDividerPosition: calc((#{var(--KDropdownItemDividerContainerHeight)} / 2 + 1px) * -1);
    position: relative;
    margin-top: var(--KDropdownItemDividerContainerHeight);

    &:before {
      position: relative;
      display: block;
      content: '';
      height: 1px;
      width: 100%;
      top: var(--KDropdownItemDividerPosition);
      background: var(--grey-200);
    }
  }

  svg {
    margin-right: .75rem;
  }

  &:hover {
    background-color: var(--grey-100);
  }

  .k-dropdown-item-trigger {
    text-align: left;
    padding: var(--spacing-md) var(--spacing-lg);
    text-decoration: none;
    width: 100%;
    color: var(--black-70);

    &:disabled,
    &.disabled {
      cursor: not-allowed !important;
      color: var(--grey-400) !important;

      &:hover {
        background-color: var(--grey-200) !important;
      }
    }
  }

  &.danger {
    button:not(:disabled),
    a:not(:disabled) {
      color: var(--red-500);
      transition: all 300ms;

      &:hover {
        color: var(--red-500);
      }
    }
  }
}
</style>

<style lang="scss">
.k-dropdown-item {
  a, button {
    &.k-dropdown-item-trigger {
      text-decoration: none !important;
    }
  }
}
</style>
