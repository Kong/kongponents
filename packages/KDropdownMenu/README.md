# @kongponents/kdropdownmenu

[![](https://img.shields.io/npm/v/@kongponents/kdropdownmenu.svg?style=flat-square)](https://www.npmjs.com/package/@kongponents/kdropdownmenu)

```html
<KDropdownMenu :description="'hello world'">
  Hello from a slot
</KDropdownMenu>
```
