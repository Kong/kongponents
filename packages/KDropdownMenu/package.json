{
  "name": "@kongponents/kdropdownmenu",
  "version": "6.36.3",
  "description": "Dropdown Menu component",
  "main": "dist/KDropdownMenu.umd.min.js",
  "componentName": "KDropdownMenu",
  "source": "index.js",
  "files": [
    "dist",
    "*.vue",
    "index.js"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Kong/kongponents.git"
  },
  "author": "Kong Inc",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/Kong/kongponents/issues"
  },
  "homepage": "https://github.com/Kong/kongponents#readme",
  "publishConfig": {
    "access": "public"
  },
  "peerDependencies": {
    "vue-router": "^3.5.4"
  },
  "dependencies": {
    "@kongponents/kbutton": "^6.36.3",
    "@kongponents/kooltip": "^6.36.3",
    "@kongponents/kpop": "^6.36.3",
    "@kongponents/ktoggle": "^6.36.3",
    "@kongponents/styles": "^6.36.3"
  }
}
