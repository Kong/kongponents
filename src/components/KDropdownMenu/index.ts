// This file is a shim to support deprecated components while properly mapping to the new component.
// If this deprecated component is used, it will warn the user to upgrade to the new component in the browser console.

import { defineComponent, h } from 'vue'
import KDropdown from '@/components/KDropdown.vue'

export default defineComponent({

  name: 'KDropdownMenu',
  setup(props, { attrs, slots, emit }) {
    console.warn("The Kongponents 'KDropdownMenu' component is deprecated and will be removed in a future release.\nPlease update all references of 'KDropdownMenu' to 'KDropdown'.\nKongponent Docs: https://alpha--kongponents.netlify.app/guide/migrating-to-version-9.html#kdropdownmenu")

    return () => h(KDropdown, {
      ...props,
      ...attrs,
      ...emit,
    }, slots)
  },
})
