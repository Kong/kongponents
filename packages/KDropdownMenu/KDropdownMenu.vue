<template>
  <div class="k-dropdown">
    <KToggle v-slot="{toggle, isToggled}">
      <KPop
        v-bind="boundKPopAttributes"
        :on-popover-click="() => {
          toggle();
          return isToggled
        }"
        @opened="toggle"
        @closed="toggle"
      >
        <div class="d-flex">
          <slot
            :is-open="isToggled"
            name="default"
          >
            <KButton
              v-if="btnText"
              class="k-dropdown-btn"
              appearance="primary"
              @click="toggle"
            >
              {{ btnText }}
            </KButton>
          </slot>
        </div>
        <template #content>
          <ul class="k-dropdown-list dropdown-list">
            <slot
              :items="items"
              :is-open="isToggled"
              name="items"
            >
              <KDropdownItem
                v-for="item in items"
                :key="item.label"
                :item="item"
              />
            </slot>
          </ul>
        </template>
      </KPop>
    </KToggle>
  </div>
</template>

<script>
import KButton from '@kongponents/kbutton/KButton.vue'
import KPop from '@kongponents/kpop/KPop.vue'
import KToggle from '@kongponents/ktoggle/KToggle'
import KDropdownItem from './KDropdownItem.vue'

const defaultKPopAttributes = {
  hideCaret: true,
  popoverClasses: 'k-dropdown-popover mt-1',
  popoverTimeout: 0,
  positionFixed: true
}

export default {
  name: 'KDropdownMenu',
  components: {
    KButton,
    KDropdownItem,
    KPop,
    KToggle
  },
  props: {
    btnText: {
      type: String,
      default: ''
    },
    // kpopAttributes is used to pass properties directly to the wrapped KPop component.
    // Commonly-overridden properties include:
    // - width
    // - placement
    // - popoverClasses
    // - target
    kpopAttributes: {
      type: Object,
      default: () => ({})
    },
    items: {
      type: Array,
      default: () => []
    }
  },
  data () {
    return {
      boundKPopAttributes: {
        ...defaultKPopAttributes,
        ...this.kpopAttributes,
        popoverClasses: `${defaultKPopAttributes.popoverClasses} ${this.kpopAttributes.popoverClasses}`
      }
    }
  }
}
</script>

<style lang="scss" scoped>
@import '~@kongponents/styles/variables';

.k-dropdown {
  .drodpown-trigger:after {
    display: inline-block;
    width: 0;
    height: 0;
    margin-left: var(--spacing-xs, spacing(xs));
    vertical-align: middle;
    content: "";
    border-top: 0.325em solid;
    border-right: 0.325em solid transparent;
    border-left: 0.325em solid transparent;
  }

  .k-dropdown-list.dropdown-list {
    min-width: 148px;
  }
}
</style>

<style lang="scss">
@import '~@kongponents/styles/variables';

.k-popover.k-dropdown-popover {
  --KPopPaddingY: var(--spacing-sm);
  --KPopPaddingX: 0;
  border: 1px solid var(--black-10);

  ul {
    margin: 0;
    padding: 0;
  }

  a {
    flex: 1;
    color: var(--black-70);

    &:hover,
    &:active,
    &:focus {
      text-decoration: none;
    }
  }
}
</style>
