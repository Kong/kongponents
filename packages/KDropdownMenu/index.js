import KDropdown from './KDropdown'
import KDropdownItem from './KDropdownItem'

const Plugin = {
  install (Vue) {
    Vue.component('KDropdown', KDropdown)
    Vue.component('KDropdownItem', KDropdownItem)
  }
}

export default Plugin
export { KDropdown, KDropdownItem }

if (typeof window !== 'undefined' && window.Vue) {
  window.Vue.use(Plugin)
}
