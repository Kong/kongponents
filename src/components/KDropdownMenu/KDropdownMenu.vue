<template>
  <div
    class="k-dropdown k-dropdown-menu"
    :class="{ 'selection-dropdown-menu': appearance === 'selectionMenu' }"
  >
    <KToggle v-slot="{ toggle, isToggled }">
      <KPop
        v-bind="boundKPopAttributes"
        data-testid="k-dropdown-menu-popover"
        :hide-popover="hidePopover"
        :on-popover-click="() => handleTriggerToggle(isToggled, toggle, false)"
        :test-mode="!!testMode || undefined"
        @closed="() => handleTriggerToggle(isToggled, toggle, false)"
        @opened="() => handleTriggerToggle(isToggled, toggle, true)"
      >
        <component
          :is="tooltipComponent"
          class="k-dropdown-trigger dropdown-trigger"
          data-testid="k-dropdown-trigger"
          :label="disabledTooltip"
          :max-width="!!disabledTooltip ? '240' : undefined"
          :position="!!disabledTooltip ? 'bottom' : undefined"
          :position-fixed="!!disabledTooltip ? true : undefined"
          :test-mode="!!testMode || undefined"
        >
          <slot
            :is-open="isToggled.value"
            name="default"
          >
            <!-- Must wrap in div to allow tooltip when disabled -->
            <div>
              <KButton
                v-if="label || icon"
                :appearance="appearance === 'selectionMenu' ? 'outline' : buttonAppearance"
                :caret-color="caretColor"
                class="k-dropdown-btn"
                data-testid="k-dropdown-btn"
                :disabled="disabled"
                :icon="icon"
                :show-caret="showCaret || appearance === 'selectionMenu'"
              >
                {{ label }}
              </KButton>
            </div>
          </slot>
        </component>
        <template #content>
          <ul
            class="k-dropdown-list dropdown-list"
            data-testid="k-dropdown-list"
          >
            <slot
              :close-dropdown="handleCloseDropdown"
              :handle-selection="handleSelection"
              :items="items"
              name="items"
            >
              <KDropdownItem
                v-for="(item, idx) in items"
                v-bind="item"
                :key="`${item.label}-${idx}`"
                :item="item"
                :selection-menu-child="appearance === 'selectionMenu'"
                @change="handleSelection"
              />
            </slot>
          </ul>
        </template>
      </KPop>
    </KToggle>
  </div>
</template>

<script lang="ts" setup>
import type { PropType, Ref } from 'vue'
import { computed, onMounted, ref, watch, nextTick } from 'vue'
import type { Appearance, ButtonAppearance, DropdownItem, PopPlacements } from '@/types'
import { AppearanceArray } from '@/types'
import KButton from '@/components/KButton/KButton.vue'
import KTooltip from '@/components/KTooltip/KTooltip.vue'
import KPop from '@/components/KPop/KPop.vue'
import KToggle from '@/components/KToggle'
import KDropdownItem from './KDropdownItem.vue'

const props = defineProps({
  appearance: {
    type: String as PropType<Appearance>,
    default: 'menu',
    validator: (value: Appearance) => AppearanceArray.includes(value),
  },
  buttonAppearance: {
    type: String as PropType<ButtonAppearance>,
    default: 'primary',
  },
  caretColor: {
    type: String,
    default: undefined,
  },
  label: {
    type: String,
    default: '',
  },
  icon: {
    type: String,
    default: '',
  },
  showCaret: {
    type: Boolean,
    default: false,
  },
  width: {
    type: String,
    default: '',
  },
  // kpopAttributes is used to pass properties directly to the wrapped KPop component.
  // Commonly-overridden properties include:
  // - placement
  // - popoverClasses
  // - target
  kpopAttributes: {
    type: Object,
    default: null,
  },
  items: {
    type: Array as PropType<Array<DropdownItem>>,
    default: () => [],
    validator: (items: DropdownItem[]) => !items.length || items.every(i => i.label !== undefined),
  },
  disabled: {
    type: Boolean,
    default: false,
  },
  disabledTooltip: {
    type: String,
    default: '',
  },
  /**
   * Test mode - for testing only, strips out generated ids
   */
  testMode: {
    type: Boolean,
    default: false,
  },
})

const emit = defineEmits<{
  (e: 'toggleDropdown', value: boolean): void;
  (e: 'change', value: DropdownItem): void;
}>()

const hidePopover = ref<boolean>(false)

const tooltipComponent = computed(() => props.disabledTooltip ? KTooltip : 'div')

const defaultKPopAttributes = {
  hideCaret: true,
  popoverClasses: 'k-dropdown-popover',
  popoverTimeout: 0,
  positionFixed: true,
  placement: 'bottomStart' as PopPlacements,
}

const boundKPopAttributes = {
  ...defaultKPopAttributes,
  ...props.kpopAttributes,
  width: props.width ? props.width : undefined,
  popoverClasses: `${defaultKPopAttributes.popoverClasses} ${props.kpopAttributes?.popoverClasses || ''}`,
}

const selectedItem = ref<DropdownItem>()

const handleSelection = (item: DropdownItem): void => {
  if (props.appearance !== 'selectionMenu') {
    return
  }
  selectedItem.value = item
}

const handleCloseDropdown = async (): Promise<void> => {
  hidePopover.value = true

  // reset the hidePopover value so it's ready for the next time the dropdown is opened
  // need nextTick to ensure the popover is hidden before resetting the value
  await nextTick(() => {
    hidePopover.value = false
  })
}

const handleTriggerToggle = (isToggled: Ref<boolean>, toggle: () => void, isOpen: boolean): boolean => {
  // avoid toggling twice for the same event
  if (isToggled.value !== isOpen) {
    toggle()
    emit('toggleDropdown', isToggled.value)
  }

  return isToggled.value
}

watch(selectedItem, (newVal, oldVal): void => {
  if (newVal && newVal !== oldVal) {
    emit('change', newVal)
  }
})

onMounted(() => {
  if (props.items) {
    const selectionArr = props.items.filter(item => item.selected)

    if (selectionArr.length) {
      selectedItem.value = selectionArr[0]
    }
  }
})
</script>

<style lang="scss" scoped>
.k-dropdown-menu {
  width: fit-content;
}
</style>

<style lang="scss">
@import '@/styles/tmp-variables';

.k-popover.k-dropdown-popover {
  border: var(--kui-border-width-10, $kui-border-width-10) solid $tmp-color-black-10;
  margin-top: var(--kui-space-20, $kui-space-20) !important;
  padding: var(--kui-space-50, $kui-space-50) var(--kui-space-0, $kui-space-0);

  ul {
    margin: var(--kui-space-0, $kui-space-0);
    padding: var(--kui-space-0, $kui-space-0);
  }

  a {
    color: var(--kui-color-text, $kui-color-text);
    flex: 1;

    &:hover,
    &:active,
    &:focus {
      text-decoration: none;
    }
  }
}

.selection-dropdown-menu {
  .dropdown-trigger .k-button {
    border: var(--kui-border-width-0, $kui-border-width-0);
    color: var(--kui-color-text-neutral-strong, $kui-color-text-neutral-strong);
    white-space: nowrap;

    &:focus {
      box-shadow: none;
    }

    &:active:disabled {
      background-color: var(--kui-color-background, $kui-color-background);
    }

    &.is-active {
      background-color: var(--kui-color-background-neutral-weakest, $kui-color-background-neutral-weakest);
    }
  }

  .k-popover.k-dropdown-popover {
    z-index: 10000 !important;

    li {
      .non-visual-button {
        font-weight: var(--kui-font-weight-regular, $kui-font-weight-regular) !important;
      }

      &.k-dropdown-selected-option {
        background-color: var(--kui-color-background-primary-weakest, $kui-color-background-primary-weakest);

        .non-visual-button {
          font-weight: var(--kui-font-weight-medium, $kui-font-weight-medium) !important;
        }
      }
    }
  }
}
</style>
