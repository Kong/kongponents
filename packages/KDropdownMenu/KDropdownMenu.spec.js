import { mount } from '@vue/test-utils'
import KDropdownMenu from '@/KDropdownMenu/KDropdownMenu'

describe('KDropdownMenu', () => {
  it('matches snapshot', () => {
    const wrapper = mount(KDropdownMenu)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
