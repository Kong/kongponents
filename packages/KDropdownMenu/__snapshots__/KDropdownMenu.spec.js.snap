// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KDropdownMenu allows slotting content 1`] = `
<div class="k-dropdown k-dropdown-menu">
  <div id="test-target-id-1234" aria-controls="test-popover-id-1234" role="button" data-testid="k-dropdown-menu-popover">
    <div label="" data-testid="k-dropdown-trigger" class="k-dropdown-trigger dropdown-trigger"><button>Click Me!</button></div>
    <div id="test-popover-id-1234" role="region" class="k-popover k-dropdown-popover mt-1 undefined hide-caret" style="width: 200px; max-width: 350px; max-height: auto; display: none;" name="fade">
      <!---->
      <div class="k-popover-content">
        <ul data-testid="k-dropdown-list" class="k-dropdown-list dropdown-list"><span>I am slotted baby!</span></ul>
      </div>
      <!---->
    </div>
  </div>
</div>
`;

exports[`KDropdownMenu matches snapshot 1`] = `
<div class="k-dropdown k-dropdown-menu">
  <div id="test-target-id-1234" aria-controls="test-popover-id-1234" role="button" data-testid="k-dropdown-menu-popover">
    <div label="" data-testid="k-dropdown-trigger" class="k-dropdown-trigger dropdown-trigger">
      <div>
        <!---->
      </div>
    </div>
    <div id="test-popover-id-1234" role="region" class="k-popover k-dropdown-popover mt-1 undefined hide-caret" style="width: 200px; max-width: 350px; max-height: auto; display: none;" name="fade">
      <!---->
      <div class="k-popover-content">
        <ul data-testid="k-dropdown-list" class="k-dropdown-list dropdown-list"></ul>
      </div>
      <!---->
    </div>
  </div>
</div>
`;

exports[`KDropdownMenu renders disabled props when passed 1`] = `
<div class="k-dropdown k-dropdown-menu">
  <div id="test-target-id-1234" aria-controls="test-popover-id-1234" role="button" data-testid="k-dropdown-menu-popover">
    <div id="3160e4b1-136d-11ed-9ddc-a1b90be1e26f" aria-controls="3160e4b0-136d-11ed-9ddc-a1b90be1e26f" role="button" class="k-dropdown-trigger dropdown-trigger">
      <div><button type="button" class="k-button k-dropdown-btn medium rounded primary" disabled="disabled" data-testid="k-dropdown-btn">
          <!---->
          Click me
          <!----></button></div>
      <div id="3160e4b0-136d-11ed-9ddc-a1b90be1e26f" role="region" class="k-popover kooltip mt-2  hide-caret" style="width: auto; max-width: 240px; max-height: auto; display: none;" name="fade">
        <!---->
        <div class="k-popover-content">
          <div role="tooltip">A sweet tooltip</div>
        </div>
        <!---->
      </div>
    </div>
    <div id="test-popover-id-1234" role="region" class="k-popover k-dropdown-popover mt-1 undefined hide-caret" style="width: 200px; max-width: 350px; max-height: auto; display: none;" name="fade">
      <!---->
      <div class="k-popover-content">
        <ul data-testid="k-dropdown-list" class="k-dropdown-list dropdown-list">
          <li data-testid="k-dropdown-item-Props" class="k-dropdown-item w-100" label="Props">
            <div data-testid="k-dropdown-item-trigger" class="k-dropdown-item-trigger">Props</div>
          </li>
          <li data-testid="k-dropdown-item-Slots" class="k-dropdown-item w-100" label="Slots">
            <div data-testid="k-dropdown-item-trigger" class="k-dropdown-item-trigger">Slots</div>
          </li>
          <li data-testid="k-dropdown-item-Top" class="k-dropdown-item w-100" label="Top">
            <div data-testid="k-dropdown-item-trigger" class="k-dropdown-item-trigger">Top</div>
          </li>
        </ul>
      </div>
      <!---->
    </div>
  </div>
</div>
`;

exports[`KDropdownMenu renders props when passed 1`] = `
<div class="k-dropdown k-dropdown-menu">
  <div id="test-target-id-1234" aria-controls="test-popover-id-1234" role="button" data-testid="k-dropdown-menu-popover">
    <div label="" data-testid="k-dropdown-trigger" class="k-dropdown-trigger dropdown-trigger">
      <div><button type="button" class="k-button k-dropdown-btn medium rounded primary" data-testid="k-dropdown-btn">
          <!---->
          Drop it!
          <!----></button></div>
    </div>
    <div id="test-popover-id-1234" role="region" class="k-popover k-dropdown-popover mt-1 undefined hide-caret" style="width: 200px; max-width: 350px; max-height: auto; display: none;" name="fade">
      <!---->
      <div class="k-popover-content">
        <ul data-testid="k-dropdown-list" class="k-dropdown-list dropdown-list">
          <li data-testid="k-dropdown-item-Props" class="k-dropdown-item w-100" label="Props">
            <div data-testid="k-dropdown-item-trigger" class="k-dropdown-item-trigger">Props</div>
          </li>
          <li data-testid="k-dropdown-item-Slots" class="k-dropdown-item w-100" label="Slots">
            <div data-testid="k-dropdown-item-trigger" class="k-dropdown-item-trigger">Slots</div>
          </li>
          <li data-testid="k-dropdown-item-Top" class="k-dropdown-item w-100" label="Top">
            <div data-testid="k-dropdown-item-trigger" class="k-dropdown-item-trigger">Top</div>
          </li>
        </ul>
      </div>
      <!---->
    </div>
  </div>
</div>
`;

exports[`KDropdownMenu renders with selected item 1`] = `
<div class="k-dropdown k-dropdown-menu">
  <div id="test-target-id-1234" aria-controls="test-popover-id-1234" role="button" data-testid="k-dropdown-menu-popover">
    <div label="" data-testid="k-dropdown-trigger" class="k-dropdown-trigger dropdown-trigger">
      <div>
        <!---->
      </div>
    </div>
    <div id="test-popover-id-1234" role="region" class="k-popover k-dropdown-popover mt-1 undefined hide-caret" style="width: 200px; max-width: 350px; max-height: auto; display: none;" name="fade">
      <!---->
      <div class="k-popover-content">
        <ul data-testid="k-dropdown-list" class="k-dropdown-list dropdown-list">
          <li data-testid="k-dropdown-item-Label-1" class="k-dropdown-item w-100 k-dropdown-selected-option" label="Label 1" value="label1">
            <div data-testid="k-dropdown-item-trigger" class="k-dropdown-item-trigger">Label 1</div>
          </li>
          <li data-testid="k-dropdown-item-US-(United States)" class="k-dropdown-item w-100" label="US (United States)" value="us">
            <div data-testid="k-dropdown-item-trigger" class="k-dropdown-item-trigger">US (United States)</div>
          </li>
          <li data-testid="k-dropdown-item-FR-(France)" class="k-dropdown-item w-100" label="FR (France)" value="fr">
            <div data-testid="k-dropdown-item-trigger" class="k-dropdown-item-trigger">FR (France)</div>
          </li>
        </ul>
      </div>
      <!---->
    </div>
  </div>
</div>
`;
